### PR TITLE
feat : Add initial SDK support for Rich Messaging (ButtonActions as Attachment)

### PR DESCRIPTION
**Pull request by Shailesh351**
Specification : https://github.com/WideChat/Rocket.Chat.Android/wiki/Buttons-Inside-Attachment-Object-Spec

Test : https://github.com/WideChat/Rocket.Chat.Android/wiki/(WIP)--Rich-Messaging-Acceptance-Test

### CODE REVIEW

**Comment by rafaelks** [comment]
@Shailesh351 Could you please sign the CLA so we can review this PR?

File: core/src/main/kotlin/chat/rocket/core/internal/AttachmentAdapter.kt [hunk 1]
```diff
@@ -315,6 +363,37 @@ class AttachmentAdapter(moshi: Moshi, private val logger: Logger) : JsonAdapter<
...
+
+    private fun writeActions(writer: JsonWriter, actions: List<Action>) {
+        if (actions.isNotEmpty()) {
+            writer.name("actions")
+            writer.beginArray()
+            actions.forEach {
+                if (it is ButtonAction) {
+                    writer.beginObject()
+                    writer.name("type").value(it.type)
+                    it.text?.run { writer.name("text").value(it.text) }
```
Comment (luciofm) [inline core/src/main/kotlin/chat/rocket/core/internal/AttachmentAdapter.kt]: If the values can be nullable, you can just write null to the JSON.

Or let's use `.let` to keep consistency with the rest of the file.
`it.text?.let { writer.name("text").value(it) }`

**Comment by Shailesh351** [comment]
@luciofm Updated PR with requested changes. Please review.

Thanks!!

**Comment by luciofm** [comment]
merged with some fixes in indentation

**Pull request merged**